simulator/sender.cc

```diff
@@ -224,54 +225,64 @@ void* Sender::MasterThread(int /*unused*/) {
         packet_count_per_symbol_[comp_frame_slot][ctag.symbol_id_] = 0;
 
         size_t next_symbol_id = FindNextSymbol((ctag.symbol_id_ + 1));
+        // Set end of frame time to the time the last symbol was transmitted
+        // (now) do this before the delay until the end of the frame
+        if (next_symbol_id == cfg_->Frame().NumTotalSyms()) {
+          double timeus_now = GetTime::GetTimeUs();
+          this->frame_end_[(ctag.frame_id_ % kNumStatsFrames)] = timeus_now;
+        }
+
         unsigned symbol_delay = next_symbol_id - ctag.symbol_id_;
-        if (kDebugPrintSender == true) {
-          std::printf("Sender: Finishing symbol %d : %zu : %zu delayed %d\n",
-                      ctag.symbol_id_, cfg_->Frame().NumTotalSyms(),
-                      next_symbol_id, symbol_delay);
+        if (kDebugPrintSender) {
+          std::printf(
+              "Sender: Finishing symbol %d, Next Symbol: %zu, Total Symbols: "
+              "%zu, delaying %d\n",
+              ctag.symbol_id_, next_symbol_id, cfg_->Frame().NumTotalSyms(),
+              symbol_delay);
         }
         // Add inter-symbol delay
         DelayTicks(tick_start, GetTicksForFrame(ctag.frame_id_) * symbol_delay);
+        tick_start = GetTime::Rdtsc();
 
         size_t next_frame_id = ctag.frame_id_;
         // Check to see if the current frame is finished
         assert(next_symbol_id <= cfg_->Frame().NumTotalSyms());
         if (next_symbol_id == cfg_->Frame().NumTotalSyms()) {
-          if ((kDebugSenderReceiver == true) ||
-              (kDebugPrintPerFrameDone == true)) {
-            std::printf("Sender: Transmitted frame %u in %.2f ms\n",
-                        ctag.frame_id_,
-                        (GetTime::GetTime() - start_time) / 1000.0);
-            start_time = GetTime::GetTime();
-          }
           next_frame_id++;
-          if (next_frame_id == cfg_->FramesToTest()) {
-            break;  // Finished
-          }
-          this->frame_end_[(ctag.frame_id_ % kNumStatsFrames)] =
-              GetTime::GetTime();
-          // Add inter-frame delay
-          DelayTicks(GetTime::Rdtsc(), ticks_inter_frame_);
-          this->frame_start_[(next_frame_id % kNumStatsFrames)] =
-              GetTime::GetTime();
```

Comment:
Frame start and frame end are used to track the frame duration in TX side. I think it is more reasonable to have the frame start as the start of a frame (not the start of TX), and the frame end as the end of a frame before the inter-frame delay (not the end of TX).

Reply:
Please review my most recent changes.  I think I addressed this to help with your concerns.